frontend/src/pages/onboarding.tsx

```diff
@@ -2,140 +2,89 @@ import { ErrorAlert } from "@components/_shared/Alerts";
 import InviteUsersStep from "@components/onboarding-steps/InviteUsersStep";
 import OrganizationSelectionStep from "@components/onboarding-steps/OrganizationSelectionStep";
 import { OnboardingFormType } from "@schema/onboarding.schema";
 import type { GetServerSidePropsContext } from "next";
-import { getCsrfToken } from "next-auth/react";
+import { getCsrfToken, useSession } from "next-auth/react";
 import { NextSeo } from "next-seo";
 import { useRouter } from "next/router";
-import { ReactNode, useEffect, useState } from "react";
+import { ReactNode, useEffect, useState, Fragment } from "react";
 import { useForm } from "react-hook-form";
 import { match } from "ts-pattern";
+import { listGroups, followGroups } from "@utils/group";
+import { inviteUser } from "@utils/user";
+import { listOrganizations, requestOrganizationOwner, requestNewOrganization } from "@utils/organization";
 
 export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const csrfToken = await getCsrfToken(context);
+  
+  const topicsData = await listGroups({
+    type: 'topic',
+  });
+  const locationData = await listGroups({
+    type: 'geography',
+  });
+  const organizationsData = await listOrganizations({
+    input: {
+      detailed: true,
+      includeUsers: true
+    }
+  });
   return {
     props: {
-      csrfToken: await getCsrfToken(context),
+      csrfToken: csrfToken ? csrfToken : "",
+      organizationsData: organizationsData,
+      topicsData: topicsData,
+      locationData: locationData
     },
   };
 }
 
-export default function LoginPage({ csrfToken }: { csrfToken: string }) {
-  const [errorMessage] = useState<string | null>(null);
+export default function LoginPage({
+  csrfToken, 
+  organizationsData, 
+  topicsData, 
+  locationData
+}: InferGetServerSidePropsType<typeof getServerSideProps>): JSX.Element {
+  const { data: sessionData } = useSession();
+  const apiKey = sessionData?.user.apikey;
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
   const [isSmallScreen, setIsSmallScreen] = useState(false);
   const [loggingIn] = useState(false);
   const router = useRouter();
   const [disableButton, setDisableButton] = useState(false);
   const form = useForm<OnboardingFormType>();
   const { handleSubmit, watch } = form;
-
-  const orgs = [
-    { id: 1, name: "Asia Development Bank (ADB)" },
-    { id: 2, name: "CAF - Banco de desarrollo de América Latina" },
-    { id: 3, name: "Chalmers University" },
-    { id: 4, name: "European Bank for Reconstrution and Development (EBRD)" },
-    { id: 5, name: "Fabrique des Mobilités" },
-    { id: 6, name: "FIA Foundation" },
-  ];
+  const [newOrgRequest, setNewOrgRequest] = useState(false);
+  // states to track whether step was skipped or submitted
+  const [firstStepSkipped, setFirstStepSkipped] = useState(true);
+  const [secondStepSkipped, setSecondStepSkipped] = useState(true);
```

Comment:
I set them initially to `true` because if a user clicks `skip` or navigates to the next tab using the navigations on the top, the data on the tab is not submitted at the end. The data for a tab is only submitted at the end, if a users clicks `Next` on that tab.